### ISSUE TITLE
Installing from source no longer installs the share directory

### ISSUE THREAD

**Issue by zshaheen** [E3SM-Project/e3sm_diags#217]
When we had `INSTALL_PATH = os.path.join(sys.prefix, 'share/e3sm_diags/')` instead of what it is now (`INSTALL_PATH = 'share/e3sm_diags/'`) installing from source worked.

@chengzhuzhang had issues installing from source and reverting it back worked. @xylar do you recall why we had to change this? It had something to do with the `noarch` builds or something. I'll need to do further testing to reconcile these differences.

**Comment by xylar** [comment]
@zshaheen, I'm trying to find that discussion.  Please don't put it back with `os.path.join(sys.prefix, 'share/e3sm_diags/')`.  It breaks stuff when you actually install it to a conda environment because it stores the path to the environment where you're running `setup.py`, not the environment where it's supposed to end up.  It's a really bad thing to do.

**Comment by xylar** [comment]
https://github.com/E3SM-Project/e3sm_diags/pull/203#discussion_r267169227

**Comment by xylar** [comment]
If you can't figure out what's going wrong, could you post the steps you're taking where "share" is not getting copied and I can try to reproduce it (and hopefully fix it) tomorrow?

**Comment by xylar** [comment]
Are you having this problem from the current master?

**Comment by xylar** [comment]
I did the following test with `master`:
```bash
conda build --python=3.7 -c e3sm -c cdat/label/v81 conda/meta.yaml
conda create -n test_e3sm_diags --use-local -c cdat/label/v81 python=3.7 e3sm_diags
```
After that finished, everything looked good to me:
```
$ ls ~/miniconda3/envs/test_e3sm_diags/share/e3sm_diags/
acme_ne30_ocean_land_mask.nc  cosp_histogram  polar
colormaps                     e3sm_logo.png   zonal_mean_2d
control_runs                  lat_lon         zonal_mean_xy
```

Again, happy to help if you can give me instructions that reproduce the problem.

**Comment by chengzhuzhang** [comment]
@xylar   I was on our py3_support branch which was branched off from master with the old INSTALL_PATH. And I made a mistake to merge in the master. The master is all good! Thank you for testing.

**Comment by zshaheen** [comment]
Hi @xylar, as usual, thanks for the detailed feedback! :) Yeah sure, it would be great if you could try to recreate this issue.

1. Go to the `py3_support` branch. In the conda folder, create an env with `e3sm_diags_env_dev_py3.yml`.
    ```
    conda env create -f e3sm_diags_env_dev_py3.yml --name e3sm_diags_py3_try2
    ```

2. Edit `setup.py`, changing `INSTALL_PATH` to `'share/e3sm_diags/'` and install the software.

3. Go to `tests/system` and run the below. It only takes about 1 minute.
    ```
    e3sm_diags -p all_sets.py -d all_sets.cfg
    ```

4. You'll get an error like so:
    ```
    cdms2.error.CDMSError: Cannot open file /global/homes/z/zshaheen/anaconda2/envs/e3sm_diags_py3_try2/share/e3sm_diags/acme_ne30_ocean_land_mask.nc (Variable not found)
    ```

5. You will see that in the share folder for your env, there won’t be an `e3sm_diags` folder even present:
    ```
    $ ls /global/homes/z/zshaheen/anaconda2/envs/e3sm_diags_py3_try2/share
    aclocal          dv3d        genutil         glib-2.0          licenses  sip        xml
    bash-completion  et          gettext         gst-plugins-base  locale    tabset
    cdp              examples    gettext-0.19.8  gtk-doc           man       terminfo
    dbus-1           ffmpeg      ghostscript     icu               proj      udunits
    doc              fontconfig  gir-1.0         info              readline  vcsaddons
    ```

**Comment by xylar** [comment]
I was able to reproduce your problem.  The issue is precisely the step you left out of your instructions above, namely "install the software".

If you use:
```bash
python setup.py install
```
the files remain in the "egg" file in `lib/python3.6/site-packages/e3sm_diags-1.6.0-py3.6.egg` and don't get expanded and put in the share folder (nor is the `acme_diags` package directly in `lib/python3.6/site-packages`, as I would have expected).  

If, instead, you use:
```
pip install . 
```
Everything ends up where you expect.

The command recommended by conda-forge is slightly more involved but does the same thing:
```
"{{ PYTHON }} -m pip install . --no-deps -vv"
```
This is what I had in `conda/meta.yaml` but I believe this is another line that was changed back during a merge.
